pkg/services/queryhistory/database.go

```diff
@@ -9,6 +9,16 @@ import (
 	"github.com/grafana/grafana/pkg/util"
 )
 
+type Datasource struct {
+	UID string `json:"uid"`
+}
+
+type QueryHistoryDatasourceIndex struct {
```

Comment:
Would it make sense to delete the `Text datasource_uid` from `query_history` table? Isn't it redundant at this point?

Reply:
Renamed in 3d6a965da6b5e5c8cfa6de69a6b2cbca12f37e16. I called it `query_history_details` as we can use it in the future to move `queries` over there as well.

> Would it make sense to delete the Text datasource_uid from query_history table? Isn't it redundant at this point?

Not yet. Removing columns would be a breaking change (in case someone downgrades) so we should do it in Grafana 12.